src/main/java/com/yahoo/bullet/aggregations/sketches/DualSketch.java

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright 2017, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.aggregations.sketches;
+
+/**
+ * This class wraps sketches that need a separate sketch for update and union operations. It manages the metadata
+ * relating to managing the common operations relating to managing dual sketches. All sketches inheriting this must
+ * call {@link #update()} and {@link #union()} when performing the respective operations. It must also call
+ * {@link #collect()} before performing any operations that change data in the sketch such as getting metadata or
+ * results or resetting.
+ */
+public abstract class DualSketch extends Sketch {
```

Comment:
I didn't want anything in DualSketch that called collect. Just wanted it to be about the managing metadata related to how collect is idempotent. KMVSketch already does it for Theta and Tuple, so this will only affect QuantileSketch really. 

I guess right now the lowest class in the class hierarchy that has a method defined is responsible for calling collect. Maybe it's easier to understand to keep it that way?